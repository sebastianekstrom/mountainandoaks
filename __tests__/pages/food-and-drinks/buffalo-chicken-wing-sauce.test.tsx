import { render, screen } from "@testing-library/react";
import React from "react";

import BuffaloChickenWing from "pages/food-and-drinks/buffalo-chicken-wing-sauce";

describe("BuffaloChickenWing page", () => {
  it("renders title correctly", () => {
    render(<BuffaloChickenWing />);
    const title = screen.queryAllByText(/Buffalo Chicken Wing Sauce/i);
    expect(title).toBeDefined();
  });

  it("renders the ingress text", () => {
    render(<BuffaloChickenWing />);
    const ingress = screen.queryAllByText(/Get ready to rock your taste buds/i);
    expect(ingress).toBeDefined();
  });

  it("renders the correct ingredients", () => {
    render(<BuffaloChickenWing />);
    const ingredientList = screen.getByTestId("INGREDIENT_LIST");
    expect(ingredientList).toMatchSnapshot();
  });

  it("renders the method steps", () => {
    render(<BuffaloChickenWing />);
    const step = screen.queryAllByText(
      /Generously smother your crispy chicken/i,
    );
    expect(step).toBeDefined();
  });
});
