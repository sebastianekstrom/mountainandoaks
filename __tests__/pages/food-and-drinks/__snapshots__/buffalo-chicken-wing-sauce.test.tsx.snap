// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BuffaloChickenWing page renders the correct ingredients 1`] = `
<section
  class="mb-8 lg:mb-0"
  data-testid="INGREDIENT_LIST"
>
  <div>
     
  </div>
  <div
    class="flex flex-col"
  >
    <ul>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Hot Pepper Sauce (such as Frank's)
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          100
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Ketchup
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          65
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Butter
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          25
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Apple cider vinegar
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          20
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Worcestershire sauce
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          3
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Celery salt
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        >
          1
          g
        </p>
      </li>
      <li
        class="flex justify-between mb-1 transition-opacity "
      >
        <div
          class=""
        >
          <p
            class="variable font-sans text-paragraph text-regular leading-paragraph font-semibold"
          >
            Salt to taste
          </p>
        </div>
        <p
          class="variable font-sans text-paragraph text-regular leading-paragraph "
        />
      </li>
    </ul>
  </div>
</section>
`;
